anti_patterns/borrow_clone.md

```diff
@@ -0,0 +1,55 @@
+# Clone to satisfy the borrow checker
+
+## Description
+
+The borrow checker prevents Rust users from developing otherwise unsafe code by
+ensuring that either: only one mutable reference exists, or potentially many but
+all immutable references exist. If the code written does not hold true to these 
+conditions, this anti-pattern arises when the developer resolves the compiler
+error by cloning the variable.
+
+
+## Example
+
+```rust
+// define any variable
+let mut x = 5;
+
+// Borrow `x` -- but clone it first
+let y = &mut (x.clone()); 
+
+// perform some action on the borrow to prevent rust from optimizing this
+//out of existence
+*y += 1;
+
+// without the x.clone() two lines prior, this line would fail on compile as
+// x has been borrowed
+// thanks to x.clone(), x was never borrowed, and this line will run.
+println!("{}", x);
+```
```

Comment:
I think we can come up with a better example; I'm not sure if this is really a problem when your type is copy.  Let me play around with some examples and get back to you.